be/src/storage/lake/tablet_manager.cpp

```diff
@@ -492,6 +502,42 @@ StatusOr<TxnLogIter> TabletManager::list_txn_log(int64_t tablet_id, bool filter_
 }
 
 StatusOr<TabletSchemaPtr> TabletManager::get_tablet_schema(int64_t tablet_id, int64_t* version_hint) {
+// TODO: Eliminate the explicit dependency on staros worker
+#ifdef USE_STAROS
+    auto shard_info_or = g_worker->get_shard_info(tablet_id);
+    if (shard_info_or.ok()) {
+        const auto& shard_info = shard_info_or.value();
+        const auto& properties = shard_info.properties;
+        auto index_id_iter = properties.find("indexId");
+        if (index_id_iter != properties.end()) {
+            auto schema_id = std::atol(index_id_iter->second.data());
+            auto cache_key = global_schema_cache_key(schema_id);
+            auto schema = lookup_tablet_schema(cache_key);
+            if (schema != nullptr) {
+                return schema;
+            }
+            // else: Cache miss, read the schema file
+            auto schema_file_path = join_path(tablet_root_location(tablet_id), schema_filename(schema_id));
```

Comment:
why not put the schema file under meta directory?
better to implement `tablet_schema_location` function in location provider.

Reply:
I don't want to modify the GC module to handle the schema file.